### PR TITLE
Link to repos

### LINKED ISSUE #6: Link to repositories
It'd be really cool if you could also use this to link to a repository by using `@username/repository`.
/CC @dt

### PR DESCRIPTION
**Pull request by deltaidea**
**WAIT:** this won't work until https://github.com/DefinitelyTyped/DefinitelyTyped/pull/25143 is merged!

---

Feature: also link`@user/repo` to the repo page. Closes #6.

I went a bit crazy and refactored the whole thing. 🤦‍♂️ I recommend reviewing this commit-by-commit to prevent your head from exploding. If this is not OK, I'll understand, just tell me to redo without refactoring.

Don't forget to run `npm install` before testing.

Please bump the version in `package.json`, maybe even to `1.0.0`, why not 🤷‍♂️ 

Thanks @ItsPugle for suggesting this feature!

### CODE REVIEW

File: src/extension.ts
Comment (dt) [inline src/extension.ts]: Hmm... This assumes everyone names their upstream remote `origin`? Previously I think it would work even if you named it something else. What would you think of iterating over the remotes until one has a URL that matches `github.com`? Or maybe trying "origin" first and if it doesn't exist, iterating like that?

Comment (deltaidea) [reply src/extension.ts]: Good catch! 👍

Let's check `origin` first and then iterate like you said. This covers the case where I have `origin` but it's not the first one in the list for some reason.
Previously it always used the last remote in the list, I think. Or maybe the first one. I haven't really tested it with multiple remotes.

I'll fix this later, maybe tomorrow.

**Comment by ghost** [comment]
Thanks for taking the lead on this one, @deltaidea! Looking through the current prod version, I can see that ghlink doesn't require an `@` to prefix `user/repo#0` to create the link - will this also be updated with this, or will it be a bit of an anomaly?

**Comment by deltaidea** [comment]
Let's make `@` optional in `@user/repo#1`  for backward compatibility but include it in the link if you choose to use it so it looks nice. This is the current implementation in this PR.

**Comment by ghost** [comment]
Sounds like a good idea! LGTM ❤️ (after the fix, of course)

**Comment by deltaidea** [comment]
Fixed and tested locally.
Waiting for that DefinitelyTyped thing to be merged before publishing. It can take a week, unfortunately.

**Review by dt — APPROVED** [review]
LGTM!

**Comment by dt** [comment]
@deltaidea I *think* I added you as a contributor to my vsce publisher acct so you should be able to re-pubslish at will? 

alternatively, if you'd prefer to transfer ownership (if that's even possible?) I'd be happy to do that too: I'm not actually using vscode in most of my day-to-day anymore :(

**Comment by deltaidea** [comment]
@dt Thanks! Understandable, and I'm totally OK with maintaining the project under your publisher name. Looks like it's not possible to transfer ownership without user action anyway, and let's not do that. 😄 

Good luck with your new IDE (or no IDE) setup!

I'll ping you if something goes wrong once I try publishing.

**Pull request merged**
